Cannot build MuMaxPlus due to Cmake

Hello,

I'm trying to build mumaxplus on Windows following the instructions in the README. I've installed:

Visual Studio 2022 Community with Desktop development with C++

CUDA Toolkit 13.0

Miniconda3 environment with Python 3.13

I cloned the repository and created a clean build directory. When building either from VS or from the Developer Prompt, I get the following error:

` EXEC : CMake error : cmake version 4.1.0 [C:\Users\lusch\mumaxplus\build\temp.win-amd64-cpython-313\Release\src\physics\physics.vcxproj] ` 

and lots of

 ` C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(166,5): error MSB3073: Der Befehl "setlocal [C:\Users\lusch\mumaxplus\build\temp.win-amd64-cpython-313\Release\src\physics\physics.vcxproj] C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(166,5): error MSB3073: C:\Users\lusch\AppData\Local\Temp\pip-build-env-dce339i8\overlay\Lib\site-packages\cmake\data\bin\cmake.exe -E copy_if_different C:/Users/lusch/mumaxplus/build/lib.win-amd64-cpython-313 [C:\Users\lusch\mumaxplus\build\temp.win-amd64-cpython-313\Release\src\physics\physics.vcxproj] `

and in the end:

` File "C:\Users\lusch\AppData\Local\Temp\pip-build-env-dce339i8\overlay\Lib\site-packages\setuptools\_distutils\dist.py", line 1021, in run_command cmd_obj.run() ~~~~~~~~~~~^^ File "<string>", line 36, in run File "<string>", line 68, in build_extension File "C:\Users\lusch\miniconda3\envs\mumaxplus\Lib\subprocess.py", line 419, in check_call raise CalledProcessError(retcode, cmd) subprocess.CalledProcessError: Command '['cmake', '--build', '.', '--target', '_mumaxpluscpp', 'install', '--config', 'Release', '--', '/m']' returned non-zero exit status 1. [end of output] note: This error originates from a subprocess, and is likely not a problem with pip. ERROR: Failed building wheel for mumaxplus Failed to build mumaxplus error: failed-wheel-build-for-install × Failed to build installable wheels for some pyproject.toml based projects ╰─> mumaxplus  `

Can someone help me ?

--- a/src/physics/CMakeLists.txt
+++ b/src/physics/CMakeLists.txt
@@ -112,11 +112,23 @@ if (WIN32)
     file(GLOB CUDA_RUNTIME_DLLS
         "${CUDAToolkit_BIN_DIR}/cufft64_*.dll"
         "${CUDAToolkit_BIN_DIR}/curand64_*.dll"
-    )
+        "${CUDAToolkit_BIN_DIR}/x64/cufft64_*.dll"
+        "${CUDAToolkit_BIN_DIR}/x64/curand64_*.dll"
+    ) # CUDA 13.0 and higher puts them below the /x64 directory.
 
-    add_custom_command(TARGET physics POST_BUILD
-        COMMAND ${CMAKE_COMMAND} -E copy_if_different
-            ${CUDA_RUNTIME_DLLS}
-            $<TARGET_FILE_DIR:_mumaxpluscpp>
-    )
+    message(STATUS "CUDAToolkit_BIN_DIR=${CUDAToolkit_BIN_DIR}")
+    message(STATUS "CUDA_RUNTIME_DLLS=${CUDA_RUNTIME_DLLS}")
+
+    if (CUDA_RUNTIME_DLLS)
+        foreach(dll IN LISTS CUDA_RUNTIME_DLLS)
+            get_filename_component(dllname "${dll}" NAME)
+            add_custom_command(TARGET physics POST_BUILD
+                COMMAND ${CMAKE_COMMAND} -E copy_if_different
+                    "${dll}"
+                    "$<TARGET_FILE_DIR:_mumaxpluscpp>/${dllname}"
+            )
+        endforeach()
+    else()
+        message(WARNING "No CUDA cufft and curand DLLs found in ${CUDAToolkit_BIN_DIR}")
+    endif()
 endif()

